### ISSUE TITLE
[ci] Sphinx job broken due to upstream changes

### ISSUE THREAD

**Issue by marscher** [BAMWelDX/weldx#844]
We can try to fix this, but I'd prefer switching to myst first and see if this will still be an issue.

**Comment by CagtayFabry** [comment]
> We can try to fix this, but I'd prefer switching to myst first and see if this will still be an issue.

agreed, let's look into it once we are on the myst stack 👍

**Comment by marscher** [comment]
unfortunately this also happens on the myst branch.. 
```Sphinx parallel build error:
sphinx.errors.ExtensionError: Handler <function process_docstring at 0x7f03371f0820> for event 'autodoc-process-docstring' threw an exception (exception: )
```

**Comment by CagtayFabry** [comment]
do you have a dev environment where it is running? (so we can pin some dependencies for testing)

**Comment by marscher** [comment]
There you go [wxdoc-running.txt](https://github.com/BAMWelDX/weldx/files/10504293/wxdoc-running.txt)

**Comment by CagtayFabry** [comment]
Here is the environment diff between the [last passing scheduled run](https://github.com/BAMWelDX/weldx/actions/runs/3927742633/jobs/6714645088) and the [failed run](https://github.com/BAMWelDX/weldx/actions/runs/3983856241/jobs/6829527056) from this weekend

`sphinx-autodoc-typehints` jumped form `1.20` to `1.23` somehow

```diff
@@ -22,7 +22,7 @@ beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
 bidict                    0.22.1             pyhd8ed1ab_0    conda-forge
 bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
 boltons                   21.0.0             pyhd8ed1ab_0    conda-forge
-bottleneck                1.3.5            py39h2ae25f5_1    conda-forge
+bottleneck                1.3.6            py39h389d5f1_0    conda-forge
 brotli                    1.0.9                h166bdaf_8    conda-forge
 brotli-bin                1.0.9                h166bdaf_8    conda-forge
 brotlipy                  0.7.0           py39hb9d737c_1005    conda-forge
@@ -37,12 +37,10 @@ cftime                    1.6.2            py39h2ae25f5_1    conda-forge
 charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
 colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
 comm                      0.1.2              pyhd8ed1ab_0    conda-forge
-commonmark                0.9.1                      py_0    conda-forge
 contourpy                 1.0.7            py39h4b4f3f3_0    conda-forge
 cryptography              39.0.0           py39h079d5ae_0    conda-forge
 curl                      7.87.0               hdc1c0ab_0    conda-forge
 cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
-dataclasses               0.8                pyhc8e2a94_3    conda-forge
 debugpy                   1.6.5            py39h227be39_0    conda-forge
 decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
 defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
@@ -53,7 +51,6 @@ flit-core                 3.8.0              pyhd8ed1ab_0    conda-forge
 fonttools                 4.38.0           py39hb9d737c_1    conda-forge
 freetype                  2.12.1               hca18f0e_1    conda-forge
 fs                        2.4.15             pyhd8ed1ab_0    conda-forge
-future                    0.18.3             pyhd8ed1ab_0    conda-forge
 gmp                       6.2.1                h58526e2_0    conda-forge
 gmpy2                     2.1.2            py39h376b7d2_1    conda-forge
 h5py                      3.7.0           nompi_py39h817c9c5_102    conda-forge
@@ -67,7 +64,7 @@ importlib-resources       5.10.2             pyhd8ed1ab_0    conda-forge
 importlib_metadata        6.0.0                hd8ed1ab_0    conda-forge
 importlib_resources       5.10.2             pyhd8ed1ab_0    conda-forge
 ipyfilechooser            0.6.0              pyhd8ed1ab_0    conda-forge
-ipykernel                 6.20.1             pyh210e3f2_0    conda-forge
+ipykernel                 6.20.2             pyh210e3f2_0    conda-forge
 ipympl                    0.9.2              pyhd8ed1ab_0    conda-forge
 ipython                   8.8.0              pyh41d4057_0    conda-forge
 ipython_genutils          0.2.0                      py_1    conda-forge
@@ -125,23 +122,25 @@ libxcb                    1.13              h7f98852_1004    conda-forge
 libxml2                   2.10.3               h7463322_0    conda-forge
 libzip                    1.9.2                hc929e4a_1    conda-forge
 libzlib                   1.2.13               h166bdaf_4    conda-forge
-markupsafe                2.1.1            py39hb9d737c_2    conda-forge
-matplotlib-base           3.6.2            py39hf9fd14e_0    conda-forge
+markdown-it-py            2.1.0              pyhd8ed1ab_0    conda-forge
+markupsafe                2.1.2            py39h72bdee0_0    conda-forge
+matplotlib-base           3.6.3            py39he190548_0    conda-forge
 matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
+mdurl                     0.1.0              pyhd8ed1ab_0    conda-forge
 meshio                    5.3.4              pyhd8ed1ab_0    conda-forge
 mistune                   2.0.4              pyhd8ed1ab_0    conda-forge
-mpc                       1.2.1                h9f54685_0    conda-forge
+mpc                       1.3.1                hfe3b2da_0    conda-forge
 mpfr                      4.1.0                h9202a9a_1    conda-forge
 mpmath                    1.2.1              pyhd8ed1ab_0    conda-forge
 msgpack-python            1.0.4            py39hf939315_1    conda-forge
 munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
 nbclassic                 0.4.8              pyhd8ed1ab_0    conda-forge
 nbclient                  0.7.2              pyhd8ed1ab_0    conda-forge
-nbconvert                 7.2.7              pyhd8ed1ab_0    conda-forge
-nbconvert-core            7.2.7              pyhd8ed1ab_0    conda-forge
-nbconvert-pandoc          7.2.7              pyhd8ed1ab_0    conda-forge
+nbconvert                 7.2.8              pyhd8ed1ab_0    conda-forge
+nbconvert-core            7.2.8              pyhd8ed1ab_0    conda-forge
+nbconvert-pandoc          7.2.8              pyhd8ed1ab_0    conda-forge
 nbformat                  5.7.3              pyhd8ed1ab_0    conda-forge
-nbsphinx                  0.8.11             pyhd8ed1ab_0    conda-forge
+nbsphinx                  0.8.12             pyhd8ed1ab_0    conda-forge
 ncurses                   6.3                  h27087fc_1    conda-forge
 nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
 netcdf4                   1.6.2           nompi_py39hfaa66c4_100    conda-forge
@@ -153,7 +152,7 @@ numpydoc                  1.5.0              pyhd8ed1ab_0    conda-forge
 openjpeg                  2.5.0                hfec8fc6_2    conda-forge
 openssl                   3.0.7                h0b41bf4_1    conda-forge
 packaging                 23.0               pyhd8ed1ab_0    conda-forge
-pandas                    1.5.2            py39h2ad29b5_2    conda-forge
+pandas                    1.5.3            py39h2ad29b5_0    conda-forge
 pandoc                    2.19.2               h32600fe_1    conda-forge
 pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
 parso                     0.8.3              pyhd8ed1ab_0    conda-forge
@@ -184,18 +183,18 @@ python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
 python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
 python-json-logger        2.0.4              pyhd8ed1ab_0    conda-forge
 python_abi                3.9                      3_cp39    conda-forge
-pytz                      2022.7             pyhd8ed1ab_0    conda-forge
+pytz                      2022.7.1           pyhd8ed1ab_0    conda-forge
 pyyaml                    6.0              py39hb9d737c_5    conda-forge
 pyzmq                     25.0.0           py39h0be026e_0    conda-forge
 readline                  8.1.2                h0f457ee_0    conda-forge
-requests                  2.28.1             pyhd8ed1ab_1    conda-forge
+requests                  2.28.2             pyhd8ed1ab_0    conda-forge
 rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
 rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
-rich                      13.1.0             pyhd8ed1ab_0    conda-forge
+rich                      13.2.0             pyhd8ed1ab_1    conda-forge
 scipy                     1.10.0           py39h7360e5f_0    conda-forge
 semantic_version          2.10.0             pyhd8ed1ab_0    conda-forge
 send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
-setuptools                65.6.3             pyhd8ed1ab_0    conda-forge
+setuptools                66.1.1             pyhd8ed1ab_0    conda-forge
 setuptools-scm            7.1.0              pyhd8ed1ab_0    conda-forge
 setuptools_scm            7.1.0                hd8ed1ab_0    conda-forge
 six                       1.16.0             pyh6c4a22f_0    conda-forge
@@ -203,7 +202,7 @@ sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
 snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
 soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
 sphinx                    5.3.0              pyhd8ed1ab_0    conda-forge
-sphinx-autodoc-typehints  1.20.2             pyhd8ed1ab_0    conda-forge
+sphinx-autodoc-typehints  1.23.4             pyhd8ed1ab_0    conda-forge
 sphinx-copybutton         0.5.0              pyhd8ed1ab_0    conda-forge
 sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
 sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
@@ -226,7 +225,7 @@ typing_extensions         4.4.0              pyha770c72_0    conda-forge
 tzdata                    2022g                h191b570_0    conda-forge
 unicodedata2              15.0.0           py39hb9d737c_0    conda-forge
 urllib3                   1.26.14            pyhd8ed1ab_0    conda-forge
-wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
+wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
 webencodings              0.5.1                      py_1    conda-forge
 websocket-client          1.4.2              pyhd8ed1ab_0    conda-forge
 weldx                     0.6.3.dev9+g17a5c1f          pypi_0    pypi
@@ -241,4 +240,4 @@ yaml                      0.2.5                h7f98852_2    conda-forge
 zeromq                    4.3.4                h9c3ff4c_1    conda-forge
 zipp                      3.11.0             pyhd8ed1ab_0    conda-forge
 zlib                      1.2.13               h166bdaf_4    conda-forge
-zstd                      1.5.2                h6239696_4    conda-forge
\ No newline at end of file
+zstd                      1.5.2                h3eb15da_6    conda-forge
\ No newline at end of file
```

**Comment by marscher** [comment]
will be fixed by #830
